it/src/main/java/org/corfudb/universe/scenario/Scenario.java

```diff
@@ -0,0 +1,118 @@
+package org.corfudb.universe.scenario;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.universe.scenario.action.Action;
+import org.corfudb.universe.scenario.action.Action.AbstractAction;
+import org.corfudb.universe.scenario.fixture.Fixture;
+import org.corfudb.universe.scenario.spec.Spec;
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Scenario is testing Api. It uses {@link Fixture}-s and provides {@link ScenarioTestCase}-s to
+ * run unit tests.
+ * A scenario provides a factory method which creates TestCase.
+ * The TestCase represents a group of tests, each test contains one action and one spec.
+ * Scenario executes all test cases and all tests included in test cases.
+ *
+ * @param <Data> data provided by a fixture
+ * @param <T>    fixture type
+ */
+public class Scenario<Data, T extends Fixture<Data>> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00119)